config/listCheckerConfig.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"istio.io/mixer/adapters"
+	"errors"
+)
+
+type ListCheckerConfig struct {
+	AdapterConfig *adapters.AdapterConfig
```

Comment:
I would have expected three structs overall in this space:

- One struct which is the adapter-specific config block, defined by the adapter implementation. This is used to pass adapter-specific options.

- One struct which is the type-specific config block (one for all ListChecker adapters), which is used to determine fact->argument assignment for that type of adapter. I think the struct being defined here is of this variety.

- Some combo that includes the above two together, representing the total configuration space for a single adapter.

Given the above, I don't understand why this has a pointer to AdapterConfig.